hack/scripts/aws/setup_aws_accounts.sh

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env bash
+
+command -v aws >/dev/null 2>&1 || { echo >&2 "Script requires aws but it's not installed.  Aborting."; exit 1; }
+
+usage() {
+    cat <<EOF
+    usage: $0 [ OPTION ]
+    Options
+    -a        Assigned AWS Account ID 1
+    -b        Assigned AWS Account ID 2
+    -u        AWS IAM user name (Required)
+EOF
+}
+
+if ( ! getopts ":a:b:u" opt); then
+    echo ""
+    echo "    $0 requires an argument!"
+    usage
+    exit 1
+fi
+
+
+while getopts ":a:b:u" opt; do
+    case $opt in
+        a)
+            AWS_ACCOUNT_ID_1="$OPTARG"
+            ;;
+        b)
+            AWS_ACCOUNT_ID_2="$OPTARG"
+            ;;
+        u)
+            AWS_USERNAME="$OPTARG"
+            ;;
+        \?)
+            echo "Invalid option: -$OPTARG" >&2
+            usage
+            exit 1
+            ;;
+        :)
+            echo "$0 Requires an argument" >&2
+            usage
+            exit 1
+            ;;
+        esac
+    done
+
+
+if [ -z "$AWS_ACCOUNT_ID_1" ]; then
+	usage
+fi
+
+if [ -z "$AWS_ACCOUNT_ID_2" ]; then
+	usage
+fi
+
+if [ -z "$AWS_USERNAME" ]; then
+	usage
+fi
+
+check_var() {
+    if [[ -z "$2" ]]; then
+        echo "$1 is not defined. Quitting ..."
+        exit 1
+    else
+        echo "$1=$2"
+    fi
+}
+
+assume_aws_role() {
+    local account_1=$1
+    local account_2=$2
+    AWS_ASSUME_ROLE=$(aws sts assume-role --profile "$account_2" --role-arn arn:aws:iam::"$account_1":role/OrganizationAccountAccessRole --role-session-name aao-account-setup)
+    check_var "AWS_ASSUME_ROLE" "$AWS_ASSUME_ROLE"
+
+    AWS_ACCESS_KEY_ID=$(echo "${AWS_ASSUME_ROLE}" | jq -r '.Credentials.AccessKeyId')
+    AWS_SECRET_ACCESS_KEY=$(echo "${AWS_ASSUME_ROLE}" | jq -r '.Credentials.SecretAccessKey')
+    AWS_SESSION_TOKEN=$(echo "${AWS_ASSUME_ROLE}" | jq -r '.Credentials.SessionToken')
+
+    check_var "AWS_ACCESS_KEY_ID" "$AWS_ACCESS_KEY_ID"
+    export AWS_ACCESS_KEY_ID
+
+    check_var "AWS_SECRET_ACCESS_KEY" "$AWS_SECRET_ACCESS_KEY"
+    export AWS_SECRET_ACCESS_KEY
+
+    check_var "AWS_SESSION_TOKEN" "$AWS_SESSION_TOKEN"
+    export AWS_SESSION_TOKEN
+}
+
+#########################
+# Account 1 - Jump Role
+#########################
+assume_aws_role "${AWS_ACCOUNT_ID_1}" osd-staging-1
+
+# Create the sts Assume role Policy which will be applied to the new Role
+ASSUME_ROLE_POLICY_ARN=$(aws iam create-policy --policy-name STSAssumeRolePolicy --policy-document file://./setup-aws-policies/AssumeRolePolicy.json | jq '.Policy.Arn')
+# Strip quotes from policy arn
+ASSUME_ROLE_POLICY_ARN=$(echo "${ASSUME_ROLE_POLICY_ARN}" | tr -d '"')
+
+### Create the Jump Role
+TRUST_ARN_FORMAT="arn:aws:iam::277304166082:user/"${USERNAME}
```

Comment:
Move `${USERNAME}` inside the quote (`"arn:aws:....user/${USERNAME}"`). Using unquoted bash vars can lead to pretty weird side effects. Probably not in this case, but it's good practice to never have variable unquoted except when exclipitly needed.